packages/google_sign_in/google_sign_in_web/README.md

```diff
@@ -190,35 +49,43 @@ You can tell `flutter run` to listen for requests in a specific host and port wi
 
-See the [**Usage** instructions of `package:google_sign_in`](https://pub.dev/packages/google_sign_in#usage)
+This implementation returns false for `supportsAuthentication`, and will throw
+if `authenticate` is called. This is because the
+[Google Identity Services (GIS) SDK](https://developers.google.com/identity/gsi/web/guides/overview)
+only allows signing in using UI provided by the SDK.
 
-Note that the **`serverClientId` parameter of the `GoogleSignIn` constructor is not supported on Web.**
+On the web, instead of providing custom UI that calls `authenticate`, you should
+display the Widget returned by `renderButton` (from `web_only.dart`), and listen
+to `authenticationEvents` to know when the user has signed in.
 
-## Example
+### Migration from versions before 0.12
 
-Find the example wiring in the [Google sign-in example application](https://github.com/flutter/packages/blob/main/packages/google_sign_in/google_sign_in/example/lib/main.dart).
+See [Migrating from Google Sign-In](https://developers.google.com/identity/gsi/web/guides/migration)
+for information about the differences between authentication in the GIS SDK and
+the SDK used in older versions of this plugin.
 
-## API details
-
-See [google_sign_in.dart](https://github.com/flutter/packages/blob/main/packages/google_sign_in/google_sign_in/lib/google_sign_in.dart) for more API details.
-
-## Contributions and Testing
+Since the GIS SDK does _not_ manage user sessions anymore, apps that relied on
+this feature might break. If long-lived sessions are required, consider using
+some user authentication system that supports Google Sign In as a federated
+Authentication provider, like
+[Firebase Auth](https://firebase.google.com/docs/auth/flutter/federated-auth#google).
 
-Tests are crucial for contributions to this package. All new contributions should be reasonably tested.
+## Authorization
 
-**Check the [`test/README.md` file](https://github.com/flutter/packages/blob/main/packages/google_sign_in/google_sign_in_web/test/README.md)** for more information on how to run tests on this package.
+The GIS SDK does not renew authorization sessions. Once the token expires
+(after 3600 seconds), API requests will begin to fail, and you must re-request
+user authorization. For example:
 
-Contributions to this package are welcome. Read the [Contributing to Flutter Plugins](https://github.com/flutter/packages/blob/main/CONTRIBUTING.md) guide to get started.
+* `401`: Missing or invalid access token.
+* `403`: Expired access token.
 
-## Issues and feedback
+See the "Integration considerations > [UX separation for authentication and authorization](https://developers.google.com/identity/gsi/web/guides/integrate#ux_separation_for_authentication_and_authorization)
+guide" in the official GIS SDK documentation for more information about this.
 
-Please file [issues](https://github.com/flutter/flutter/issues/new)
-to send feedback or report a bug.
+### Migration from versions before 0.12
```

Comment:
Is this paragraph duplicated on purpose? IMO this could be a `##` after the Issues and feedback section, and have it there only.

Reply:
They aren't quite duplicated; authn and authz have different migration docs. See lines 44/45 in the pre-PR version of the file.

Reply:
I totally didn't notice the difference in the migration link. Too subtle after reading this far :P